### ISSUE TITLE
High Contrast Mode: Highlighted plain text unreadable

### ISSUE THREAD

**Issue by robfallows** [microsoft/vscode#90819]
Issue Type: <b>Bug</b>

Open a plain text file in high contrast mode and highlight some text.

Expected behaviour:

![image](https://user-images.githubusercontent.com/6086035/74653614-74796d00-5180-11ea-8150-d9dcd3987544.png)

Actual behaviour:

![image](https://user-images.githubusercontent.com/6086035/74653714-a8ed2900-5180-11ea-9952-6801c720c314.png)

VS Code version: Code - Insiders 1.43.0-insider (6ec6f9e3f4d17f0cef0480688a1f350b9141e567, 2020-02-17T05:42:20.509Z)
OS version: Windows_NT x64 10.0.18362

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-6700HQ CPU @ 2.60GHz (8 x 2592)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>viz_display_compositor: enabled_on<br>viz_hit_test_surface_layer: disabled_off_ok<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|15.89GB (3.08GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (60)</summary>

Extension|Author (truncated)|Version
---|---|---
markdown-toc|Ala|1.5.6
cucumberautocomplete|ale|2.14.1
Handlebars|and|0.4.1
spellright|ban|3.0.50
random-string-password-generator|bib|1.0.2
markdown-emoji|bie|0.0.9
markdown-preview-github-styles|bie|0.1.6
npm-intellisense|chr|1.3.0
vaulty|cod|0.2.0
markdown-table-prettify|dar|2.4.0
vscode-markdownlint|Dav|0.34.0
vscode-eslint|dba|2.0.15
githistory|don|0.5.3
xml|Dot|2.5.0
gitlens|eam|10.2.1
vscode-pull-request-github|Git|0.14.0
todo-tree|Gru|0.0.167
ansible|haa|0.2.8
beautify|Hoo|1.5.0
markdown-extended|jeb|1.0.16
diff-tool|jin|0.0.1
MeteorSnippets|jml|0.1.3
vscode-github|Kni|0.30.3
rpm-spec|Lau|0.2.3
AWK|lug|0.0.2
bash-ide-vscode|mad|1.4.0
terraform|mau|1.4.0
rainbow-csv|mec|1.5.0
azure-pipelines|ms-|1.165.1
vscode-azurestorage|ms-|0.8.0
vscode-azureterraform|ms-|0.3.0
vscode-docker|ms-|0.10.0
python|ms-|2020.2.63072
remote-containers|ms-|0.101.0
remote-ssh|ms-|0.49.0
remote-ssh-edit|ms-|0.49.0
remote-wsl|ms-|0.42.3
vscode-remote-extensionpack|ms-|0.19.0
azure-account|ms-|0.8.8
azurecli|ms-|0.5.0
powershell|ms-|2020.1.0
vscode-node-azure-pack|ms-|0.0.10
vsliveshare|ms-|1.0.1615
team|ms-|1.161.0
azurerm-vscode-tools|msa|0.8.4
debugger-for-chrome|msj|4.12.6
sqltools|mtx|0.21.6
vscode-paste-image|mus|1.0.4
printcode|nob|3.0.0
indent-rainbow|ode|7.4.0
text-marker|ryu|1.11.0
arm-snippets|sam|1.4.19
jinjahtml|sam|0.11.0
bash-beautify|sha|0.1.1
vscode-markdown-paste-image|tel|0.12.3
ansible-autocomplete|tim|0.0.2
vscodeintellicode|Vis|1.2.5
vscode-icons|vsc|10.0.0
vscode-ansible|vsc|0.5.2
vscode-surround|yat|1.0.2

</details>

**Comment by isidorn** [comment]
This is trully horrible :)
Imho the issue is that the selection is white in High Contrast mode. It aslo does not work that well in some language, for example Typescript - contrast is not that high.

Assigning also to @misolori to potentially find a better high contrast selection color
@aeschli might you know why we use white for High Contrast background in the first place?

![Screenshot 2020-02-18 at 11 05 13](https://user-images.githubusercontent.com/1926584/74725699-8c172b00-523e-11ea-92d9-bf173b5fec85.png)

**Comment by aeschli** [comment]
FYI @alexdima 
I agree that looks bad.

In the early days we used `mix-blend-mode: difference`
https://github.com/microsoft/vscode/commit/ab5846a2ab4a25df02997de7e894582521d5a532

We typically use borders for hc.

**Comment by isidorn** [comment]
Yeah, borders would work better here imho.

**Comment by robfallows** [comment]
The issue I have with borders is that in all the examples I've seen (e.g. git diffs) there's one border per line. That makes a multi-line highlight very noisy. If there was a way to collapse adjacent borders, that would be really cool 😄

**Comment by robfallows** [comment]
In any event, this appears to be a regression, as it works as expected in the stable release (1.42.1).

**Comment by isidorn** [comment]
Agree, it is a regression. We already have it assigned to this milestone so we plan to fix.
@aeschli for now assigning to you and you can pass it around. Thanks

**Comment by aeschli** [comment]
Selection border request is here: https://github.com/microsoft/vscode/issues/30793

**Comment by aeschli** [comment]
ollapse adjacent border code: https://github.com/microsoft/vscode/blob/e5dadf5e8a00c1b84c77ac4197804967bd393ab5/src/vs/editor/browser/viewParts/selections/selections.ts#L154

**Comment by aeschli** [comment]
@RMacfarlane @alexdima This was caused by the change for https://github.com/microsoft/vscode/issues/90386

The selection foreground color requires the 'inline-selected-text' line decoration which is set here:
https://github.com/microsoft/vscode/blob/03b30ff3cceedb2f82718fecfba58d8a4cf4e71c/src/vs/editor/browser/viewParts/lines/viewLine.ts#L199

Looks like HC was always broken when renderWhitespace === 'selection'. With the change of the default it becomes obvious.
